agent/bls.mjs

```diff
@@ -686,59 +717,70 @@ function perform_bls_verify_u256( details, joGlueResult, u256, joCommonPublicKey
+        details.write( strLogPrefix + cc.debug( "BLS u256/summary verify output is:\n" ) + cc.notice( strOutput ) + "\n" );
         details.write( strLogPrefix + cc.success( "BLS u256/summary verify success" ) + "\n" );
         fnShellRestore();
         return true;
     } catch ( err ) {
-        const s1 = strLogPrefix + cc.fatal( "BLS u256/summary verify CRITICAL ERROR:" ) + cc.normal( " error description is: " ) + cc.warning( err.toString() ) + "\n";
+        const s1 = strLogPrefix + cc.fatal( "BLS u256/summary verify CRITICAL ERROR:" ) + cc.error( " error description is: " ) + cc.warning( owaspUtils.extract_error_message( err ) ) + cc.error( ", stack is: " ) + "\n" + cc.stack( err.stack ) + "\n";
         const s2 = strLogPrefix + cc.error( "BLS u256/summary verify output is:\n" ) + cc.notice( strOutput ) + "\n";
-        log.write( s1 );
         details.write( s1 );
-        log.write( s2 );
         details.write( s2 );
         fnShellRestore();
     }
     return false;
 }
 
 async function check_correctness_of_messages_to_sign( details, strLogPrefix, strDirection, jarrMessages, nIdxCurrentMsgBlockStart, joExtraSignOpts ) {
-    let w3 = null; let joMessageProxy = null; let joAccount = null; let joChainName = null;
+    const imaState = state.get();
+    let joMessageProxy = null, joAccount = null, joChainName = null;
     if( strDirection == "M2S" ) {
-        w3 = imaState.w3_main_net;
         joMessageProxy = imaState.jo_message_proxy_main_net;
-        joAccount = imaState.joAccount_main_net;
-        joChainName = imaState.strChainName_s_chain;
+        joAccount = imaState.chainProperties.mn.joAccount;
+        joChainName = imaState.chainProperties.sc.strChainName;
     } else if( strDirection == "S2M" ) {
-        w3 = imaState.w3_s_chain;
         joMessageProxy = imaState.jo_message_proxy_s_chain;
-        joAccount = imaState.joAccount_s_chain;
-        joChainName = imaState.strChainName_main_net;
+        joAccount = imaState.chainProperties.sc.joAccount;
+        joChainName = imaState.chainProperties.mn.strChainName;
     } else if( strDirection == "S2S" ) {
-        w3 = joExtraSignOpts.w3_src;
-        joMessageProxy = new w3.eth.Contract( imaState.joAbiPublishResult_s_chain.message_proxy_chain_abi, imaState.joAbiPublishResult_s_chain.message_proxy_chain_address );
-        joAccount = imaState.joAccount_s_chain;
+        joAccount = imaState.chainProperties.sc.joAccount;
         joChainName = joExtraSignOpts.chain_id_dst;
+        const ethersProvider =
+            ( "ethersProvider_src" in joExtraSignOpts && joExtraSignOpts.ethersProvider_src )
+                ? joExtraSignOpts.ethersProvider_src
+                : null
+                ;
+        if( ! ethersProvider )
+            throw new Error( "CRITICAL ERROR: No provider specified in extra signing options for checking messages of direction \"" + strDirection + "\"" );
```

Comment:
https://github.com/skalenetwork/internal-support/blob/main/docs/wiki/code-guidelines.md#code-line-length

Reply:
Done, code line length check also added into ES-LINT settings and done for all JS code now